### PR TITLE
BUGZ-576: Removing Logspam on sit

### PR DESCRIPTION
**Pull request by PrestonB1123**
Removing logspam related to animation restore from sit.
https://highfidelity.atlassian.net/browse/BUGZ-576
also resolves
https://highfidelity.atlassian.net/browse/DEV-151

### CODE REVIEW

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 1]
```diff
@@ -113,6 +113,10 @@
 
         // Set isSitting value in Settings
         var sitNewSettings = [_this.entityID];
+        changedSeats = false;
+        if(Settings.getValue(SETTING_KEY_AVATAR_SITTING, false) !== false && Settings.getValue(SETTING_KEY_AVATAR_SITTING, false)[0] !== _this.entityID){
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]:
```suggestion
        if (Settings.getValue(SETTING_KEY_AVATAR_SITTING, false) !== false && Settings.getValue(SETTING_KEY_AVATAR_SITTING, false)[0] !== _this.entityID){
```

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 2]
```diff
@@ -178,11 +180,12 @@
         for (var i in roles) { // restore roles to prevent overlap
             // -----------------THIS IS PRINTING ERRORS WHEN IN HMD WHEN CHANGING SEATS---------------------
             // Rig::restoreRoleAnimation could not find role ***ROLE NAME***
-            MyAvatar.restoreRoleAnimation(roles[i]); 
+            if(changedSeats){
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]:
```suggestion
            if (changedSeats){
```

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 3]
```diff
@@ -291,13 +294,16 @@
         var settingsEntityID = sitCurrentSettings[0];
 
         MyAvatar.clearPinOnJoint(MyAvatar.getJointIndex("Hips"));
+        changedSeats = false;
+        Controller.actionEvent.disconnect(onActionEvent);
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: why did you move this?

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js (hunk 1 above)
Comment (PrestonB1123) [reply usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: Ack

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js (hunk 2 above)
Comment (PrestonB1123) [reply usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: Ack

Comment (zfox23) [reply usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: please fix - looks like you missed this in your latest commit (or didn't do it)

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js (hunk 1 above)
Comment (zfox23) [reply usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: please fix - looks like you missed this in your latest commit (or didn't do it)

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 4]
```diff
@@ -98,7 +97,20 @@
...
+            standUp(continueSittingDown);
+            changedSeats = true;
+        } else {
+            continueSittingDown();
+        }
+        
+    }
+
+    function continueSittingDown() {
+
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: remove extra whitespace please

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 5]
```diff
@@ -282,22 +296,29 @@
...
 
+        MyAvatar.clearPinOnJoint(MyAvatar.getJointIndex("Hips"));
+        changedSeats = false;
+        if (isConnected) {
+            Controller.actionEvent.disconnect(onActionEvent);
+            isConnected = false;
+        }
         // STANDING FROM THIS CHAIR
         // Make avatar stand up (if changed seat do not do this)
         if (settingsEntityID === _this.entityID) { // POSSIBLE RACE CONDITION WITH SETTINGS BEING CHANGED BY NEW SEAT
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: what happens if you remove this conditional entirely? now that you've (theoretically) fixed the race condition, shouldn't you not need this anymore?

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js (hunk 4 above)
Comment (PrestonB1123) [reply usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: Ack

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 6]
```diff
@@ -98,7 +97,20 @@
...
+    var changedSeats = false;
     function startSitDown() {
+        var sitEntity = Settings.getValue(SETTING_KEY_AVATAR_SITTING, false);
+        if (sitEntity && sitEntity !== _this.entityID) {
+            standUp(continueSittingDown);
+            changedSeats = true;
+        } else {
+            continueSittingDown();
+        }
+
```
[the remark this answers is not in the document]
Comment (PrestonB1123) [reply usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: Ack

File: usefulUtilities/sitV2/appResources/appData/sitClientV2.js [hunk 7]
```diff
@@ -289,15 +297,22 @@
 
         // get the entityID, previous position and orientation 
         var sitCurrentSettings = Settings.getValue(SETTING_KEY_AVATAR_SITTING);
-        var settingsEntityID = sitCurrentSettings[0];
+        var settingsEntityID = sitCurrentSettings;
 
+        MyAvatar.clearPinOnJoint(MyAvatar.getJointIndex("Hips"));
```
Comment (zfox23) [inline usefulUtilities/sitV2/appResources/appData/sitClientV2.js]: Why is this here and on line 311?
